builtin/providers/openstack/resource_openstack_fw_firewall_v1.go

```diff
@@ -60,6 +68,18 @@ func resourceFWFirewallV1() *schema.Resource {
 	}
 }
 
+// Firewall is an OpenStack firewall.
+type Firewall struct {
+	ID           string   `json:"id"`
+	Name         string   `json:"name"`
+	Description  string   `json:"description"`
+	AdminStateUp bool     `json:"admin_state_up"`
+	Status       string   `json:"status"`
+	PolicyID     string   `json:"firewall_policy_id"`
+	TenantID     string   `json:"tenant_id"`
+	RouterIDs    []string `json:"router_ids"`
```

Comment:
Ah, does the result of GET now have `router_ids`? I wasn't aware of that or else I would have dealt with it in the original gophercloud PR.

This should be fine for now, but can you move it to `types.go`?